### PR TITLE
Bump hibernate-core

### PR DESCRIPTION
**Pull request by jjestrel**
for real this time

### CODE REVIEW

File: misk-testing/src/main/kotlin/misk/testing/MiskTestExtension.kt [hunk 1]
```diff
@@ -80,7 +80,7 @@ internal class MiskTestExtension : BeforeEachCallback, AfterEachCallback {
       if (context.startService()) {
         try {
           serviceManager.startAsync().awaitHealthy(60, TimeUnit.SECONDS)
-        } catch (e: IllegalStateException) {
+        } catch (e: IllegalStateException) { log.error(e) { "Error" }
```
Comment (jjestrel) [inline misk-testing/src/main/kotlin/misk/testing/MiskTestExtension.kt]: TODO: remove

File: misk-hibernate/src/main/kotlin/misk/hibernate/SessionFactoryService.kt [hunk 2]
```diff
@@ -107,8 +106,13 @@ internal class SessionFactoryService(
 
     // NB: MetadataSources.getMetadataBuilder() returns a different instance each call!
     val metadataBuilder = metadataSources.metadataBuilder
+    // After `build` is called Hibernate creates ID Generators so we have to register this custom
+    // type before we call build, otherwise Id will be treated as a SerializableType instead.
+    // See: Component#createIdentifierGenerator
+    //metadataBuilder.applyBasicType(IdType(), Id::class.jvmName)
```
Comment (jmuia) [inline misk-hibernate/src/main/kotlin/misk/hibernate/SessionFactoryService.kt]: accidentally left commented code

File: misk-hibernate/src/main/kotlin/misk/hibernate/SessionFactoryService.kt [hunk 3]
```diff
@@ -107,8 +106,13 @@ internal class SessionFactoryService(
...
     val metadata = metadataBuilder.build() as MetadataImplementor
```
Comment (swankjesse) [inline misk-hibernate/src/main/kotlin/misk/hibernate/SessionFactoryService.kt]: would it be better to rename this to be like `draftMetadata` so the actual one gets the good name?

**Comment by swankjesse** [comment]
@jjestrel How can I help land this?

**Comment by jjestrel** [comment]
@swankjesse fixed merge conflicts. will merge tomorrow morning, get this out and give a PSA

File: misk-hibernate/src/main/kotlin/misk/hibernate/SessionFactoryService.kt (hunk 2 above)
Comment (jjestrel) [reply misk-hibernate/src/main/kotlin/misk/hibernate/SessionFactoryService.kt]: Removed

File: misk-hibernate/src/main/kotlin/misk/hibernate/SessionFactoryService.kt (hunk 3 above)
Comment (jjestrel) [reply misk-hibernate/src/main/kotlin/misk/hibernate/SessionFactoryService.kt]: Renamed

**Pull request merged**